pkg/eigenState/stakerShares/stakerShares.go

```diff
@@ -610,88 +653,172 @@ func (ss *StakerSharesModel) HandleStateChange(log *storage.TransactionLog) (int
 	return nil, nil
 }
 
+// TODO: this is not very performant in cases of large slashings and is called twice (once in CommitFinalState and once in GenerateStateRoot)
+// but prob ok?
 // prepareState prepares the state for commit by adding the new state to the existing state.
 func (ss *StakerSharesModel) prepareState(blockNumber uint64) ([]*StakerSharesDiff, error) {
-	preparedState := make([]*StakerSharesDiff, 0)
-
-	accumulatedState, ok := ss.stateAccumulator[blockNumber]
-	if !ok {
-		err := xerrors.Errorf("No accumulated state found for block %d", blockNumber)
-		ss.logger.Sugar().Errorw(err.Error(), zap.Error(err), zap.Uint64("blockNumber", blockNumber))
-		return nil, err
-	}
+	// reset the delta accumulator
+	ss.deltaAccumulator[blockNumber] = make([]*StakerShareDeltas, 0)
+	// keep track of the updated staker shares
+	updatedStakerShares := make(map[types.SlotID]*StakerSharesDiff)
 
-	slotIds := make([]types.SlotID, 0)
-	for slotId := range accumulatedState {
-		slotIds = append(slotIds, slotId)
-	}
+	for _, diff := range ss.diffAccumulator[blockNumber] {
+		switch diff.Event.(type) {
+		case *StakerShareDeltas:
```

Comment:
Reflection like this should be avoided if possible. You're better off defining a struct that can be sparsely populated with separately named and type fields. e.g. 

```go
type StateDiff struct {
	StakerShareDelta *StakerSharesDeltas
	SlashDiff *SlashDiff
}
```

Then you can do a simple if `if diff.StakerShareDelta != nil {`

Reply:
https://github.com/Layr-Labs/sidecar/pull/68/commits/849bd0db77bd553611d6ccdf776de875882dc9c5

Reply:
Let me know if this pattern looks fine, resolving other comments on the same topic for now